src/main/java/net/sixpointsix/bespoke/processautomation/calculator/restapi/controller/IndustriesController.java

```
@@ -0,0 +1,104 @@
+        params.setNonactive(nonActive);
+        List<Industry> industries= industryService.getIndustries(params);
+        IndustryListResponse response = new IndustryListResponse(industries);
+        return response;
+    }
+
+    @RequestMapping(value = "/", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseStatus(HttpStatus.CREATED)
+    public Industry insertIndustry(@Valid @RequestBody SaveIndustryRequest industry){
+        Industry savedIndustry = industryService.insertIndustry(industry);
+        return savedIndustry;
+    }
+
+    @RequestMapping(value = "/{uuid}", method = RequestMethod.PUT, consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseStatus(HttpStatus.CREATED)
+    public Industry updateIndustry(@Valid @RequestBody SaveIndustryRequest industry, @PathVariable UUID uuid) throws ChangeSetPersister.NotFoundException, OutOfDateException {
+        Industry updatedIndustry;
+        try{
+            updatedIndustry = industryService.updateIndustry(industry, uuid);
+        }catch(OutOfDateException e){
+            throw new OutOfDateException(uuid);
+        }
+
+        if(updatedIndustry==null){
+            throw new ChangeSetPersister.NotFoundException();
+        }else{
+            return updatedIndustry;
+        }
+    }
+
+    @RequestMapping(value = "/{uuid}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public DeletedItemResponse deleteIndustry(@PathVariable UUID uuid) throws ChangeSetPersister.NotFoundException, OutOfDateException{
+        DeletedItemResponse response;
+        try {
+            response = industryService.deleteIndustry(uuid);
+        }catch (OutOfDateException e){
+            throw new OutOfDateException(uuid);
+        }
+         if(response==null){
```

let the service raise the notfound exception as the problem is within that.